torch/_inductor/codegen/cpp.py

```diff
@@ -785,14 +789,25 @@ def load(self, name: str, index: sympy.Expr):
         new_index = self.transform_index(index)
 
         if expanded_index == new_index:
-            line = f"at::vec::Vectorized<float>({var}[{cexpr(index)}])"
+            if V.graph.get_dtype(name) in [torch.bool, torch.uint8, torch.float64]:
+                g_tmp_buf = f"g_tmp_buffer_{var}"
+                nelements = 1
+                if f"float {g_tmp_buf}[{nelements}] = {{0}};" not in self.loads._lines:
+                    self.loads.writeline(f"float {g_tmp_buf}[{nelements}] = {{0}};")
+                self.loads.writeline(
+                    f"to_float<double>({var}, {g_tmp_buf}, {nelements});"
+                )
+                line = f"at::vec::Vectorized<float>({g_tmp_buf}[{cexpr(index)}])"
+            else:
+                line = f"at::vec::Vectorized<float>({var}[{cexpr(index)}])"
         else:
-            if V.graph.get_dtype(name) in [torch.bool, torch.uint8]:
+            if V.graph.get_dtype(name) in [torch.bool, torch.uint8, torch.float64]:
                 g_tmp_buf = f"g_tmp_buffer_{var}"
                 nelements = codecache.pick_vec_isa().nelements()
-                self.loads.writeline(f"float {g_tmp_buf}[{nelements}] = {{0}};")
+                if f"float {g_tmp_buf}[{nelements}] = {{0}};" not in self.loads._lines:
+                    self.loads.writeline(f"float {g_tmp_buf}[{nelements}] = {{0}};")
```

Comment:
Duplicate code. Please abstract it as a common code block.

Reply:
Yes, simplified it.